scipy/special/_hyp2f1.pxd

```diff
@@ -109,17 +117,33 @@ cdef inline double complex hyp2f1_complex(
     if fabs(1 - z.real) < EPS and z.imag == 0 and c - a - b < 0:
         return NPY_INFINITY + 0.0j
     # Gauss's Summation Theorem for z = 1; c - a - b > 0 (DLMF 15.4.20).
-    # Fallback to Fortran original. To be translated to Cython later.
     if z == 1.0 and c - a - b > 0:
-        return double_complex_from_npy_cdouble(
-            chyp2f1_wrap(a, b, c, npy_cdouble_from_double_complex(z))
-        )
+        result = Gamma(c) * Gamma(c - a - b)
+        result /= Gamma(c - a) * Gamma(c - b)
+        # Try again with logs if there has been an overflow.
```

Comment:
Note that long-term the right way to do this is to use the Lanczos approximation:

https://www.boost.org/doc/libs/1_38_0/libs/math/doc/sf_and_dist/html/math_toolkit/backgrounders/lanczos.html

to combine the large terms analytically. We've ported it to Cephes here:

https://github.com/scipy/scipy/blob/master/scipy/special/cephes/lanczos.c

Reply:
Nice! That looks really useful. I'll do this in my next PR.